File: Easy/majority_element/majority_element.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
    public:
        int majorityElement(vector<int>& nums) {
            int count = 0;
            int candidate = 0;

            for(int i = 0; i < nums.size(); i++){
                if(count == 0){
                    candidate = nums[i];
                }
                if(nums[i] == candidate){
                    count++;
                } else {
                    count--;
                }
            }
            return candidate;
        }
};

//Example usage:
int main(){
    Solution solution;
    vector<int> nums1 = {3, 2, 3};
    int result1 = solution.majorityElement(nums1);
    cout << "For the input [3, 2, 3], the majority element is: " << result1 << endl;

    vector<int> nums2 = {2, 2, 1, 1, 1, 2, 2};
    int result2 = solution.majorityElement(nums2);
    cout << "For the input [2, 2, 1, 1, 1, 2, 2], the majority element is: " << result2 << endl;

    return 0;
}

File: Easy/majority_element/majority_element.py
from typing import List

def majorityElement(nums: List[int]) -> int:
    count = 0
    for num in nums:
        if count == 0:
            majority = num
        if num == majority:
            count += 1
        else:
            count -= 1
    return majority

if __name__ == "__main__":
    print("Testing majorityElement function:")
    nums1 = [3,2,3]
    print("For the input nums1[3, 2, 3], the majority element is:", majorityElement(nums1)) #Output: 3

    nums2 = [2,2,1,1,1,2,2]
    print("For the input nums2[2, 2, 1, 1, 1, 2, 2], the majority element is:", majorityElement(nums2)) #Output: 2
